Assignment5.java

```diff
@@ -0,0 +1,198 @@
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.LinkedList;
+import java.util.LinkedHashSet;
+import java.util.Collections;
+
+public class Assignment5 {
+    /**
+     * Infers the alphabet of an unknown language from list of words of that language.
+     *
+     * @param words list of words in lexicographic order.
+     * @return characters in lexicographic order.
+     */
+    public static List<Character> inferAlphabet(List<String> words) {
+        if (words.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Set<Character> set = new HashSet<>();
```

Comment:
You can also add all the characters to the graph instead of to the set and then add the edge between the vertices. If you don't use set variable you don't need to check lines 101 - 107 in "createEdgeBetweenVertices" and line 49 in "inferAlphabet".